bahamas-client-challenge/src/main/java/com/rupeal/challenge/bahamas/client/services/InvoiceRegistryService.java

```diff
@@ -0,0 +1,63 @@
+package com.rupeal.challenge.bahamas.client.services;
+
+import com.rupeal.challenge.bahamas.client.model.ClientEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Mocked external invoice client registry class. Only Logs possible request.
+ */
+@ApplicationScoped
+public class InvoiceRegistryService {
+    private static final Logger LOG = LoggerFactory.getLogger(InvoiceRegistryService.class);
+
+    private final Client client;
+    private final InvoiceRegistryProperties properties;
+
+    public InvoiceRegistryService() {
+        this(new InvoiceRegistryProperties());
+    }
+
+    @Inject
+    public InvoiceRegistryService(final InvoiceRegistryProperties properties) {
+        this.properties = properties;
+        ExecutorService executorService = Executors.newCachedThreadPool();
+        client = ClientBuilder.newBuilder().executorService(executorService).build();
+    }
+
+
+    public void registerInvoice(final Long invoiceId, final ClientEntity invoiceClient) {
+        LOG.info("Submitting client registry to {}{}", properties.getBaseUrl(), buildParameterString(invoiceId, invoiceClient));
+
+        //Example code for the actual request if it was not a mock.
+        /*final Response response = client.target(properties.getBaseUrl())
```

Comment:
Instead of commenting the code what other approaches could you take to mock the calls?

Reply:
Thank you for the question.
I was actually just going leave this bean with the Logging only. But then I was curious as to how the actual call would be done, so I tried doing it and then commented it.
One possible way I can think of right now, is using the @Alternatives of CDI. With this, I could have the actual bean with the actual call as well as an Alternative bean with the Logged message only. Then in the beans.xml add the alternative bean so it's the "Mocked"/Alternative to be injected instead of the real one.